### DISCUSSION TITLE
Support for external wlan adapters?

### DISCUSSION THREAD

**Discussion by larsgrobe** [Freetz-NG/freetz-ng#447] [Allgemein] [answered]
Dear all,

I am considering to install a Fritzbox 7390 as a router. Unfortunately, I can get my uplink only via wlan, and the built-in hardware achieves rather poor connection quality. I have a recent wlan adapter available that comes with an external antenna and may give better results (Zyxel NWD6605). Can I use such a USB-connected wlan adapter as an uplink with Freetz?

Best, Lars.

**Comment by fda77** [comment, marked as answer]
You need the kernel modules, with 7390 its good that the kernel could be replace. But its a very old version, maybe the drivers where not available yet.
Additional you need tools to configure it (SSID, password, etc), im not sure if they are availabel with freetz (you could add them by yourself)
You need also to know that Fritzos is no "normal" linux, but uses closed source stuff for simple things, like dns or routing. Additional there is a HW paket-accelerator which handles traffic and the kernel does not know about it. Maybe you need to disable it and have less throughput
